pymeasure/instruments/wentworthlabs/s200.py

```diff
@@ -0,0 +1,548 @@
+# THE SOFTWARE.
+#
+import logging
+import time
+from enum import IntEnum
+
+from pymeasure.instruments import Instrument
+from pymeasure.instruments.validators import strict_discrete_set, strict_range
+from pymeasure.instruments.validators import multivalue_strict_range
+
+log = logging.getLogger(__name__)
+
+
+# log.addHandler(logging.NullHandler())
+
+
+class S200(Instrument):
+    """ Represents the Wenthworth Labs S200 Probe Table
+    and provides a high-level for interacting with the instrument.
+
+    .. code-block:: python
+
+    s200 = S200("GPIB0::28::INSTR", query_delay=0.5, write_delay=0.5)
+    # test probe table lamp
+    s200.lamp_on = True
+    s200.lamp_on = False
+    # test chuck lift
+    s200.chuck_lift = True
+    s200.chuck_lift = False
+    # configure x and y index
+    s200.x_index = 2000
+    s200.y_index = 2000
+    # test the moving of the chuck by x_index and y_index increment
+    s200.next_die = 'U'
+    s200.next_die = 'D'
+    s200.next_die = 'L'
+    s200.next_die = 'R'
+    # test the theta rotation of the chuck
+    s200.chuck_lift = False
+    s200.move_to_probing_zone_centre_position()
+    for i in range(0, 100):
+        s200.theta_position = -(i * 100)
+    s200.move_to_manual_load_position()
+    """
+
+    # Movement boundaries of the Pegassus S200 chuck
+    S200_MAX_X = 210000  # in microns
+    S200_MAX_Y = 210000  # in microns
+    S200_MIN_X = 1100  # in microns
+    S200_MIN_Y = 1100  # in microns
+
+    Z_OVERTRAVEL_VALID_RANGE = [0, 100]  # in microns
+    Z_FINELIFT_VALID_RANGE = [0, 10000]  # in microns
+    Z_GROSSLIFT_VALID_RANGE = [0, 100000]  # in microns
+    X_Y_INDEX_VALID_RANGE = [0, 10000]  # in tens of microns
+    X_POS_VALID_RANGE = [S200_MIN_X, S200_MAX_X]  # in tens of microns
+    Y_POS_VALID_RANGE = [S200_MIN_Y, S200_MAX_Y]  # in tens of microns
+
+    def __init__(self,
+                 adapter,
+                 name="Wentworth Labs S200 Probe Table",
+                 write_termination="\n",
+                 read_termination="\n",
+                 query_delay=0.5,
+                 write_delay=0.5,
+                 timeout=5000,
+                 **kwargs):
+        super().__init__(
+            adapter,
+            name,
+            write_termination=write_termination,
+            read_termination=read_termination,
+            send_end=True,
+            includeSCPI=True,
+            timeout=timeout,
+            **kwargs
+        )
+        self.write_delay = write_delay
+        self.query_delay = query_delay
+        self.last_write_timestamp = 0.0
+        self.command_execution_info = S200.ExecutionInfoCode(0)
+
+    chuck_lift = Instrument.setting(
+        "%s",
+        "Control the chuck to the fine down (CDW) or fine up (CUP) position.",
+        # The GDW/GUP command is used to move the chuck to the Gross Up or Gross
+        # Down positions.
+
+        validator=strict_discrete_set,
+        values={True: 'CUP', False: 'CDW'},
```

Comment:
`chuck_lift = True` would seem to indicate the `check_lift` is enabled, not that it will move up or down in a direction.

`values={'up': 'CUP', 'down': 'CDW'}` 
or
`values={'fine_up': 'CUP', 'fine_down': 'CDW'}`

Would one of those make more sense for procedure code?

Reply:
If you want to move something up or down, methods are better: one for moving upwards, one for downwards.

Reply:
Thanks to both of you. I thing I will create methods.

Reply:
Many thanks to both of you!!!